issuetracker/internal/youtrack/youtrack_test.go

```diff
@@ -0,0 +1,35 @@
+package youtrack
+
+import (
+	"fmt"
+	"testing"
+)
+
+func Test_IssueTracker_IssueURLFor(t *testing.T) {
+	var tests = []struct {
+		input  string
+		taskID string
+		want   string
+	}{
+		{"https://youtrack.com/n/projects/1", "taskId", "https://youtrack.com/api/issues/taskId?fields=customFields(value(isResolved))"},
+		{"youtrack.myjetbrains.com/n/projects/1-2_3", "taskId", "https://youtrack.myjetbrains.com/youtrack/api/issues/taskId?fields=customFields(value(isResolved))"},
+		{"yOUtrack.com/projects/1-2_3", "taskId", "https://youtrack.com/api/issues/taskId?fields=customFields(value(isResolved))"},
+		{"httPS://youtrack.myjetbrains.com/n/projects/1-2_3", "taskId", "https://youtrack.myjetbrains.com/youtrack/api/issues/taskId?fields=customFields(value(isResolved))"},
+		{"https://www.youtrack.com/n/projects/2459511", "taskId", "https://www.youtrack.com/api/issues/taskId?fields=customFields(value(isResolved))"},
+		{"http://www.youtrack.com/n/projects/2459511", "taskId", "http://www.youtrack.com/api/issues/taskId?fields=customFields(value(isResolved))"},
```

Comment:
Could you add test cases for hosted youtrack instances?

E.g.
```
youtrack.pmihaylov.com/n/projects...
http://youtrack.pmihaylov.com/n/projects...
https://youtrack.pmihaylov.com/n/projects...
myyoutrack.com/n/projects...
http://myyoutrack.com/n/projects...
https://myyoutrack.com/n/projects...
localhost:8080/n/projects...
http://locallyhosted.com:8080/n/projects...
https://locallyhosted.com:8080/n/projects...
localhost:8080/n/projects...
```

Reply:
The URLs in tests which don't include ".myjetbrains.com" are for hosted YouTrack instances. I can refactor the tests for readability but these cases are already tested.

Reply:
Ah okay, then in that case it would be nice if it's obvious which test cases are for self hosted instances by renaming the examples to e.g.:
```
selfhosted.com/...
```
Also, could you add tests for handling instances which are hosted at a specific port, different from :80. E.g.
```
myinstance.selfhosted.com:12345/...
```

Reply:
Tests are refactored.